Alignment/HIPAlignmentAlgorithm/test/testHIPAnalyzers.cc

```diff
@@ -0,0 +1,78 @@
+#include "FWCore/TestProcessor/interface/TestProcessor.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+
+// Function to run the catch2 tests
+//___________________________________________________________________________________________
+void runTestForAnalyzer(const std::string& baseConfig, const std::string& analyzerName) {
+  edm::test::TestProcessor::Config config{baseConfig};
+
+  SECTION(analyzerName + " base configuration is OK") { REQUIRE_NOTHROW(edm::test::TestProcessor(config)); }
+
+  SECTION(analyzerName + " No Runs data") {
+    edm::test::TestProcessor tester(config);
+    REQUIRE_NOTHROW(tester.testWithNoRuns());
+  }
+
+  SECTION(analyzerName + " beginJob and endJob only") {
+    edm::test::TestProcessor tester(config);
+    REQUIRE_NOTHROW(tester.testBeginAndEndJobOnly());
+  }
+
+  // the following part is commented because the
+  // HIPTwoBodyDecayAnalyzer crashes on
+  // No "TransientTrackRecord" record found in the EventSetup.
+
+  /* 
+  SECTION("No event data") {
+   edm::test::TestProcessor tester(config);
+   REQUIRE_NOTHROW(tester.test());
```

Comment:
alas this fails with :

```
----- Begin Fatal Exception 19-Jun-2024 13:45:55 CEST-----------------------
An exception of category 'NoRecord' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path '_test_endpath'
   [2] Calling method for module HIPTwoBodyDecayAnalyzer/'trackAnalyzer'
Exception Message:
No "TransientTrackRecord" record found in the EventSetup.

 The Record is delivered by an ESSource or ESProducer but there is no valid IOV for the synchronization value.
 Please check
   a) if the synchronization value is reasonable and report to the hypernews if it is not.
   b) else check that all ESSources have been properly configured.
----- End Fatal Exception -------------------------------------------------
```

Despite I explicitly add the `ESProducer` in the `catch2` configuration:

https://github.com/cms-sw/cmssw/blob/0eec2cb3f17cb5a9f0cdddc319e8a5a7c84046b1/Alignment/HIPAlignmentAlgorithm/test/testHIPAnalyzers.cc#L57-L58

any help / hints are welcome.